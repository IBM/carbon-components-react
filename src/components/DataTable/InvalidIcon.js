import PropTypes from 'prop-types';
import React from 'react';
import { iconErrorSolid } from 'carbon-icons';
import Tooltip from '../Tooltip';

/**
 * The form validation UI specific to table batch editing.
 */
const InvalidIcon = ({ children, ...other }) => (
  <Tooltip icon={iconErrorSolid} triggerText={null} {...other}>
    {children}
  </Tooltip>
);

InvalidIcon.propTypes = {
  /**
   * The child nodes.
   */
  children: PropTypes.node,
};

export default InvalidIcon;
