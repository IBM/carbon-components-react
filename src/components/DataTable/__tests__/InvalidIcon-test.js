import React from 'react';
import { shallow } from 'enzyme';
import InvalidIcon from '../InvalidIcon';

describe('InvalidIcon', () => {
  it('should render', () => {
    const wrapper = shallow(<InvalidIcon />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render custom attributes on the top-level node', () => {
    const wrapper = shallow(<InvalidIcon data-foo="Foo" />);
    expect(wrapper.prop('data-foo')).toBe('Foo');
  });

  it('should render custom child nodes', () => {
    const wrapper = shallow(
      <InvalidIcon>
        <div />
        <span />
      </InvalidIcon>
    );
    expect(wrapper.find('div').length).toBe(1);
    expect(wrapper.find('span').length).toBe(1);
  });
});
