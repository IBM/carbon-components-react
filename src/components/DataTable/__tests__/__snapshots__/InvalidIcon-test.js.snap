// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InvalidIcon should render 1`] = `
<Tooltip
  direction="bottom"
  icon={
    Object {
      "height": "16",
      "id": "icon--error--solid",
      "name": "icon--error--solid",
      "styles": "",
      "svgData": Object {
        "circles": "",
        "ellipses": "",
        "paths": Array [
          Object {
            "d": "M8 16A8 8 0 1 1 8 0a8 8 0 0 1 0 16zM3.293 4.707l8 8 1.414-1.414-8-8-1.414 1.414z",
          },
        ],
        "polygons": "",
        "polylines": "",
        "rects": "",
      },
      "tags": "icon--error--solid",
      "viewBox": "0 0 16 16",
      "width": "16",
    }
  }
  iconDescription="tooltip"
  menuOffset={[Function]}
  open={false}
  showIcon={true}
  triggerText={null}
/>
`;
